### DISCUSSION TITLE
Rich-text Editor: custom language types for Markdown syntax highlighting

### DISCUSSION THREAD

**Discussion by mashakos** [tinacms/tinacms#5273] [Ideas]
Would be great if, in the rich-text editor, we could type in any programming language type we like for a code block. Since this is regular Markdown you can technically do this when you switch to raw MD editing, but for non-technical editors - would be a nice to have.
![image](https://github.com/user-attachments/assets/43d73ded-8689-4beb-b2cf-e7fc3ab0671f)

**Comment by bradystroud** [comment]
Do you mean add search to the language selector? I agree, that would be a nice improvement.

**Reply by brookjeynes-ssw** [reply]
I believe they want a way to provide their own languages to the selector. For example, the code box Tina provides does not include JSX so you have to manually add the language tag via raw md.
